compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java

```diff
@@ -66,12 +67,76 @@ public enum DiagnosticsCodes {
     PERSIST_502("PERSIST_502", "an identity field cannot be nillable", ERROR),
     PERSIST_503("PERSIST_503", "only ''int'', ''string'', ''float'', ''boolean'', ''decimal'' " +
-            "types are supported as identity fields, found ''{0}''", ERROR);
+            "types are supported as identity fields, found ''{0}''", ERROR),
+    PERSIST_600("PERSIST_600", "invalid use of Mapping annotation. mapping name cannot be empty.", ERROR),
+    PERSIST_601("PERSIST_601", "mapping name is same as model definition.", WARNING),
+    PERSIST_602("PERSIST_602", "invalid use of Mapping annotation. " +
+            "Mapping annotation cannot be used for relation fields.", ERROR),
+    PERSIST_604("PERSIST_604", "invalid use of ''{0}'' annotation. " +
+            "''{0}'' annotation can only be used for string data type.", ERROR),
+    PERSIST_605("PERSIST_605", "invalid use of VarChar and Char annotations. " +
+            "only one of either Char or Varchar annotations can be used at a time.", ERROR),
+    PERSIST_606("PERSIST_606", "invalid use of Decimal annotation. " +
+            "Decimal annotation can only be used for decimal data type.", ERROR),
+    PERSIST_607("PERSIST_607", "invalid use of ''{0}'' annotation. " +
+            "length cannot be 0.", ERROR),
+    PERSIST_608("PERSIST_608", "invalid use of Decimal annotation. " +
+            "precision cannot be 0.", ERROR),
+    PERSIST_609("PERSIST_609", "invalid use of Decimal annotation. " +
+            "precision cannot be less than scale.", ERROR),
+    PERSIST_610("PERSIST_610", "invalid use of Mapping annotation. " +
+            "duplicate mapping name found.", ERROR),
+    PERSIST_611("PERSIST_611", "invalid use of Index annotation. " +
+            "Index annotation cannot be used for relation fields.", ERROR),
+    PERSIST_612("PERSIST_612", "invalid use of UniqueIndex annotation. " +
+            "UniqueIndex annotation cannot be used for relation fields.", ERROR),
+    PERSIST_613("PERSIST_613", "invalid use of Index annotation. " +
+            "duplicate index names.", ERROR),
+    PERSIST_614("PERSIST_614", "invalid use of UniqueIndex annotation. " +
+            "duplicate index names.", ERROR),
+    PERSIST_615("PERSIST_615", "invalid use of Index annotation. " +
+            "there cannot be empty index names.", ERROR),
+    PERSIST_616("PERSIST_616", "invalid use of UniqueIndex annotation. " +
+            "there cannot be empty index names.", ERROR),
+    PERSIST_617("PERSIST_617", "invalid use of Generated annotation. " +
+            "generated annotation can only be used for readonly fields.", ERROR),
+    PERSIST_618("PERSIST_618", "invalid use of Generated annotation. " +
+            "partial key fields cannot be auto-generated.", ERROR),
+    PERSIST_619("PERSIST_619", "invalid use of Generated annotation. " +
+            "a generated field can only be an integer data type.", ERROR),
```

Comment:
```suggestion
    PERSIST_619("PERSIST_619", "invalid use of the `Generated` annotation. " +
            "a generated field can only be an integer data type.", ERROR),
```

Reply:
This was fixed in https://github.com/ballerina-platform/module-ballerina-persist/pull/274/commits/3c315874e5c75f11888145ab8925a64a6fa4e6d9